api/v1alpha1/externaldns_types.go

```diff
@@ -421,37 +432,22 @@ type ExternalDNSSourceUnion struct {
 	// +required
 	Type ExternalDNSSourceType `json:"type"`
 
-	// AnnotationFilter describes an annotation filter
-	// used to filter which source instance resources
-	// ExternalDNS publishes records for.
-	// The annotation filter uses label selector semantics
-	// against source resource annotations.
+	// LabelFilter specifies a label selector used to filter which source instance resources ExternalDNS publishes
+	// records for. The filter uses label selector semantics against source resource labels.
 	//
 	// +optional
-	AnnotationFilter map[string]string `json:"annotationFilter,omitempty"`
-
-	// Namespace instructs ExternalDNS to only acknowledge
-	// source resource instances in a specific namespace.
-	//
-	// +optional
-	Namespace *string `json:"namespace,omitempty"`
+	LabelFilter *metav1.LabelSelector `json:"labelFilter,omitempty"`
 
 	// Service describes source configuration options specific
 	// to the service source resource.
 	//
 	// +optional
 	Service *ExternalDNSServiceSourceOptions `json:"service,omitempty"`
 
-	// CRD describes source configuration options specific
-	// to the CRD source resource.
-	// See
-	// https://github.com/kubernetes-sigs/external-dns/blob/master/docs/contributing/crd-source.md
-	// for more information about the requirements for ExternalDNS Source CRD requirements.
-	//
+	// OpenShiftRoute source configuration options for specifying ingress controller names.
+	// +kubebuilder:validation:Optional
 	// +optional
-	CRD *ExternalDNSCRDSourceOptions `json:"crd,omitempty"`
-
-	// OpenShiftRoute source currently has no unique configuration options.
+	OpenShiftRoute *ExternalDNSOpenShiftRouteOptions `json:"openshiftRouteOptions,omitempty"`
```

Comment:
Since this an openshift repo, this seems ok, but do you have a sense for how you will handle requests for extending this to other types?